nixio/dimensions.py

```diff
@@ -313,3 +315,81 @@ def labels(self):
     def labels(self, labels):
         dt = util.vlen_str_dtype
         self._h5group.write_data("labels", labels, dtype=dt)
+
+
+class DataFrameDimension(Dimension):
+
+    def __init__(self, h5group, index):
+        super(DataFrameDimension, self).__init__(h5group, index)
+
+    @classmethod
+    def _create_new(cls, parent, index, data_frame, column):
+        """
+        Create a new Dimension that points to a DataFrame
+        :param parent: DataArray the dimension will be attached to
+        :param data_frame: the referenced DataFrame for this Dimension
+        :param column: the index of a column in the DataFrame that the Dimension will reference (optional)
+        :return: The new DataFrameDimension
+        """
+        newdim = super(DataFrameDimension, cls)._create_new(parent, index)
+        newdim.data_frame = data_frame
+        newdim.column = column
+        newdim._dimension_type = DimensionType.DataFrame
+        return newdim
+
+    def get_unit(self, index=None):
```

Comment:
I think the public methods should have docstrings. Especially if using it with the default argument can lead to a ValueError.